# Exploratory Data Analysis: Purple Air

Fall 2020

Julia Longmate, Hikari Murayama, Michelle Sims, Marshall Worsham

In [2]:
import pandas as pd

## About the Data
*Retrieval Link*: [Link](https://docs.google.com/document/d/15ijz94dXJ-YAZLi9iZ_RaBwrZ4KtYeCy08goGBwnbCU/edit)

*Description*: Purple Air sells sensors to anyone who wants to buy one! They collect data from each of these individually owned sensors, and have a network of monitors world wide.

## Brief Description
### Sensor location

In [3]:
sensor_loc = pd.read_csv('data/purple_sensors_all.csv')
sensor_loc.head()

Unnamed: 0     ID                         Label DEVICE_LOCATIONTYPE  \
0           0  14633             Hazelwood canary              outside   
1           2  25999   Villages of Bridgestone AQI             outside   
2           6  53069                            #2             outside   
3           8  16931                       #SFAQ10             outside   
4          10  22751                #ValleyClimate             outside   

   THINGSPEAK_PRIMARY_ID THINGSPEAK_PRIMARY_ID_READ_KEY  \
0                 559921               CU4BQZZ38WO5UJ4C   
1                 694803               OO5PFS7JTQQSHQHE   
2                1042611               XA27RZLBWBYVQ4G0   
3                 592265               1ENJG2P25X9OVDRN   
4                 655100               2E6LYRAOFLQWQMIH   

   THINGSPEAK_SECONDARY_ID THINGSPEAK_SECONDARY_ID_READ_KEY        Lat  \
0                   559922                 D0YNZ1LM59LL49VQ  37.275561   
1                   694804                 LW93B1YLLZ4N4QFM  30.053808   
2                  1042612                 2RV70ZAU41ZVX1P8  47.190197   
3                   592267                 8QT4N8YSSJ8M2LVD  37.722417   
4                   655101                 GUIQIBMAGN82R0Z2  36.807377   

          Lon  ...  temp_f  pressure    AGE  \
0 -121.964134  ...    87.0   1011.99      1   
1  -95.494643  ...   101.0   1003.56      0   
2 -122.177992  ...    63.0    996.94      1   
3 -122.439245  ...    75.0    990.62  40233   
4 -119.796109  ...    82.0   1007.14      0   

                                               Stats  ParentID  Flag  A_H  \
0  {"v":2.55,"v1":2.38,"v2":1.99,"v3":1.67,"v4":0...       NaN   NaN  NaN   
1  {"v":30.4,"v1":28.61,"v2":27.5,"v3":27.81,"v4"...       NaN   1.0  NaN   
2  {"v":7.72,"v1":4.9,"v2":3.33,"v3":2.89,"v4":2....       NaN   NaN  NaN   
3  {"v":2.82,"v1":1.41,"v2":3.09,"v3":5.52,"v4":7...       NaN   1.0  NaN   
4  {"v":1.67,"v1":1.63,"v2":1.61,"v3":1.77,"v4":2...       NaN   NaN  NaN   

   Voc  Ozone1                       geometry  
0  NaN     NaN  POINT (-121.964134 37.275561)  
1  NaN     NaN   POINT (-95.494643 30.053808)  
2  NaN     NaN  POINT (-122.177992 47.190197)  
3  NaN     NaN  POINT (-122.439245 37.722417)  
4  NaN     NaN  POINT (-119.796109 36.807377)  

[5 rows x 26 columns]

In [4]:
sensor_loc.shape

(5945, 26)

In [5]:
sensor_loc['DEVICE_LOCATIONTYPE'].unique()

array(['outside'], dtype=object)

We have 5945 outside sensors (only downloaded outside).

## Example data

In [7]:
sensor_data = pd.read_csv('data/19245.csv')
sensor_data.head()

Unnamed: 0                     date  pm1_0  pm2_5  pm10_x  uptime_minutes  \
0           0  2018-11-15 00:00:00 UTC  27.17  59.43   72.57           66.88   
1           1  2018-11-16 00:00:00 UTC  23.56  47.52   57.80           51.15   
2           2  2018-11-17 00:00:00 UTC    NaN    NaN     NaN             NaN   
3           3  2018-11-18 00:00:00 UTC    NaN    NaN     NaN             NaN   
4           4  2018-11-19 00:00:00 UTC    NaN    NaN     NaN             NaN   

    rssi  temp_f  humidity_pct  pm2_5.1    um0_3    um0_5     um1  um2_5  \
0 -48.56   75.66         24.22    40.70  4978.82  1503.59  484.71  60.69   
1 -56.79   64.21         27.87    32.25  3776.70  1132.39  336.42  41.99   
2    NaN     NaN           NaN      NaN      NaN      NaN     NaN    NaN   
3    NaN     NaN           NaN      NaN      NaN      NaN     NaN    NaN   
4    NaN     NaN           NaN      NaN      NaN      NaN     NaN    NaN   

     um5  um10    pm1  pm10_y  
0  17.16  5.92  17.16    5.92  
1  12.24  3.56  12.24    3.56  
2    NaN   NaN    NaN     NaN  
3    NaN   NaN    NaN     NaN  
4    NaN   NaN    NaN     NaN

In [8]:
sensor_data.describe()

Unnamed: 0       pm1_0       pm2_5      pm10_x  uptime_minutes  \
count  553.000000  349.000000  349.000000  349.000000       349.00000   
mean   276.000000    4.146390    6.899370    8.023152     17783.14212   
std    159.781622    4.181046    7.389501    8.426983     17111.75390   
min      0.000000    0.010000    0.090000    0.160000         6.43000   
25%    138.000000    1.590000    2.650000    3.370000      5392.00000   
50%    276.000000    3.210000    5.110000    5.840000     12151.26000   
75%    414.000000    4.950000    8.140000    9.930000     24207.00000   
max    552.000000   27.760000   59.430000   72.570000     70629.00000   

             rssi      temp_f  humidity_pct     pm2_5.1        um0_3  \
count  349.000000  349.000000    349.000000  349.000000   349.000000   
mean   -85.386132   66.795845     51.175158    6.658968   899.848911   
std      3.581510    8.213393     12.038900    6.401104   715.092389   
min    -94.580000   50.590000     13.300000    0.050000    89.990000   
25%    -87.160000   60.440000     44.600000    2.650000   466.170000   
50%    -86.030000   66.580000     53.570000    5.140000   737.370000   
75%    -83.670000   72.640000     59.250000    8.140000  1067.240000   
max    -48.560000   92.140000     74.120000   40.700000  5022.210000   

             um0_5         um1       um2_5         um5        um10  \
count   349.000000  349.000000  349.000000  349.000000  349.000000   
mean    255.300516   46.452779    4.372292    1.442550    0.581404   
std     207.893736   50.742349    5.458284    1.547538    0.536693   
min      24.220000    2.910000    0.280000    0.110000    0.050000   
25%     127.270000   19.230000    1.790000    0.640000    0.280000   
50%     208.100000   32.380000    3.000000    1.070000    0.450000   
75%     299.170000   56.270000    5.280000    1.810000    0.730000   
max    1503.590000  484.710000   60.690000   17.160000    5.920000   

              pm1      pm10_y  
count  349.000000  349.000000  
mean     4.019112    7.600745  
std      3.748406    6.973335  
min      0.010000    0.120000  
25%      1.570000    3.370000  
50%      3.210000    5.700000  
75%      4.950000    9.790000  
max     24.210000   49.210000

Purple air provides with many different parameters related to pm and um, as well as temperature and humidity. I wish this had a quality parameter, but otherwise I don't find anything particularly suprising. The other caveat is that most of these sensors are closer to urban areas.